cloud/scope/powervs_cluster_test.go

```diff
@@ -868,166 +758,337 @@ func TestGetLoadBalancerState(t *testing.T) {
-		name         string
-		expectedLB   *infrav1beta2.VPCLoadBalancerSpec
-		clusterScope PowerVSClusterScope
-	}{
-		{
-			name: "IBMPowerVSCluster spec is empty",
-			clusterScope: PowerVSClusterScope{
-				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
-			},
-			expectedLB: &infrav1beta2.VPCLoadBalancerSpec{
-				Name:   "-loadbalancer",
-				Public: ptr.To(true),
-			},
-		},
-		{
-			name: "one public loadbalancer is configured",
-			clusterScope: PowerVSClusterScope{
-				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
-					Spec: infrav1beta2.IBMPowerVSClusterSpec{
-						LoadBalancers: []infrav1beta2.VPCLoadBalancerSpec{
-							{
-								Name:   "lb",
-								Public: ptr.To(true),
-							},
+		clusterScope := PowerVSClusterScope{
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					LoadBalancers: []infrav1beta2.VPCLoadBalancerSpec{
+						{
+							Name:   "lb1",
+							Public: core.BoolPtr(false),
+						},
+						{
+							Name:   "lb2",
+							Public: core.BoolPtr(true),
+						},
+					},
+				},
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					LoadBalancers: map[string]infrav1beta2.VPCLoadBalancerStatus{
+						"lb1": {
+							Hostname: ptr.To("lb1-hostname"),
+						},
+						"lb2": {
+							Hostname: ptr.To("lb2-hostname"),
 						},
 					},
 				},
 			},
-			expectedLB: &infrav1beta2.VPCLoadBalancerSpec{
-				Name:   "lb",
-				Public: ptr.To(true),
+		}
+
+		hostName, err := clusterScope.GetPublicLoadBalancerHostName()
+		g.Expect(hostName).To(Equal(ptr.To("lb2-hostname")))
+		g.Expect(err).To(BeNil())
+	})
+	t.Run("Multiple public load balancer names are set in IBMPowerVSCluster spec", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					LoadBalancers: []infrav1beta2.VPCLoadBalancerSpec{
+						{
+							Name:   "lb1",
+							Public: core.BoolPtr(true),
+						},
+						{
+							Name:   "lb2",
+							Public: core.BoolPtr(true),
+						},
+					},
+				},
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					LoadBalancers: map[string]infrav1beta2.VPCLoadBalancerStatus{
+						"lb1": {
+							Hostname: ptr.To("lb1-hostname"),
+						},
+						"lb2": {
+							Hostname: ptr.To("lb2-hostname"),
+						},
+					},
+				},
 			},
-		},
-		{
-			name: "multiple public loadbalancer is configured",
-			clusterScope: PowerVSClusterScope{
-				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
-					Spec: infrav1beta2.IBMPowerVSClusterSpec{
-						LoadBalancers: []infrav1beta2.VPCLoadBalancerSpec{
-							{
-								Name:   "lb",
-								Public: ptr.To(true),
-							},
-							{
-								Name:   "lb2",
-								Public: ptr.To(true),
-							},
+		}
+
+		hostName, err := clusterScope.GetPublicLoadBalancerHostName()
+		g.Expect(hostName).To(Equal(ptr.To("lb1-hostname")))
```

Comment:
Hope it wont flake with order mismatch?

Reply:
No, it will return the first public load balancer as expected.